src/eth_provider/database/ethereum.rs

```diff
@@ -356,4 +359,185 @@ mod tests {
             Some(mock_pending_transaction.tx)
         );
     }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_ethereum_block_store() {
+        // Initialize MongoDB fuzzer
+        let mut mongo_fuzzer = MongoFuzzer::new(RANDOM_BYTES_SIZE).await;
+
+        // Start MongoDB Docker container
+        let _c = DOCKER_CLI.run(mongo_fuzzer.mongo_image());
+
+        // Mock a database with 100 transactions, receipts, and headers
+        let database = mongo_fuzzer.mock_database(100).await;
+
+        // Generate random bytes for test data
+        let bytes: Vec<u8> = (0..RANDOM_BYTES_SIZE).map(|_| rand::random()).collect();
+        let mut unstructured = arbitrary::Unstructured::new(&bytes);
+
+        // Test fetching existing and none existing header via blockhash and blocknumber from database
+        test_get_header(&database, &mongo_fuzzer).await;
+
+        // Test fetching existing and none existing block via blockhash and blocknumber from database
+        test_get_blocks(&database, &mongo_fuzzer, &mut unstructured).await;
+
+        // Test fetching existing and none existing transaction counts via blockhash and blocknumber from database
+        test_get_transaction_count(&database, &mongo_fuzzer).await;
+    }
+
+    async fn test_get_header(database: &Database, mongo_fuzzer: &MongoFuzzer) {
+        let header_block_hash = mongo_fuzzer
+            .documents()
+            .get(&CollectionDB::Headers)
+            .unwrap()
+            .first()
+            .unwrap()
+            .extract_stored_header()
+            .unwrap()
+            .header
+            .clone();
+
+        // Test retrieving header by block hash
+        assert_eq!(database.header(header_block_hash.hash.unwrap().into()).await.unwrap().unwrap(), header_block_hash);
+
+        // Test retrieving header by block number
+        assert_eq!(
+            database.header(header_block_hash.number.unwrap().into()).await.unwrap().unwrap(),
+            header_block_hash
+        );
+
+        let mut rng = rand::thread_rng();
+        // Test retrieving non-existing header by block hash
+        assert_eq!(database.header(B256::from(U256::from(rng.gen::<u64>())).into()).await.unwrap(), None);
```

Comment:
Here why not using `B256::random()` instead of the complicate line to generate the random hash?